Include/internal/pycore_stackref.h

```diff
@@ -8,106 +8,130 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
+#include "pycore_object_deferred.h"
+
 #include <stddef.h>
 
 typedef union {
     uintptr_t bits;
 } _PyStackRef;
 
-static const _PyStackRef Py_STACKREF_NULL = { .bits = 0 };
 
+#define Py_TAG_INT      (0)
 #define Py_TAG_DEFERRED (1)
+#define Py_TAG_RESERVED (2)
+#define Py_TAG_PTR      (3)
+
+#define Py_TAG          (Py_TAG_PTR)
+
+static const _PyStackRef Py_STACKREF_NULL = { .bits = 0 | Py_TAG_DEFERRED};
+
+static inline int
+PyStackRef_IsNull(_PyStackRef stackref)
+{
+    return stackref.bits == Py_STACKREF_NULL.bits;
+}
+
+static inline int
+PyStackRef_IsDeferred(_PyStackRef ref)
+{
+    return ((ref.bits & Py_TAG) == Py_TAG_DEFERRED);
+}
 
 // Gets a PyObject * from a _PyStackRef
-#if defined(Py_GIL_DISABLED)
 static inline PyObject *
-PyStackRef_Get(_PyStackRef tagged)
+PyStackRef_To_PyObject_Borrow(_PyStackRef tagged)
```

Comment:
These names do not look like other CPython function naming conventions to me. Can we do something like:

* `PyStackRef_AsPyObjectBorrow(_PyStackRef)` -- instead of this funcion

And, for example:

* `PyStackRef_AsPyObjectSteal`
* `PyStackRef_FromPyObjectBorrow`
* `PyStackRef_FromPyObjectSteal`